### DISCUSSION TITLE
How to implement a "Smart" WiFi reset ?

### DISCUSSION THREAD

**Discussion by ygelt** [arendst/Tasmota#22647] [Q&A]
Hi all,
I use athom ESP32-based Tasmota relay boards that must be installed in various places, so I need an easy way to reset their WiFi to connect them even without WiFiConfig command.
Ideally by pushing a button 10 seconds or smthg like that on boot, which would display the WiFi config page. (the Tasmota "recover" method of  7 on/off is too complicated for my friends).
For info I use a TasmoCompiler-compiled "ESP32 Generic" Tasmota binary because I do not use Berry nor rules but only the Tasmota script langage.
Anyone could help ?

**Comment by barbudor** [comment]
You may have missed the same reset feature is also available by pressing Button1 for 40 seconds

Now if you don't want to reset all settings but just restart the device in AP mode for 2 minutes, you can add a rule to catch a long press on Button1 and issue a 'WifiConfig 2'

**Reply by barbudor** [reply]
And if don't have rules but scripting language, the same should be possible but I have no idea of the syntax

**Reply by sfromis** [reply]
The "factory reset" with 40 seconds may indeed be "too harsh" for Wifi config, but if you have a gpio configured as a "Button", 6 short "clicks" on that will start Wi-Fi manager. No rules nor scripting needed. Of course, this is quite insecure, and you can disable that option, but it does get closer to what you ask for.

Anyway, all ESP32 builds should include Berry, which is a much better option for managing things, and can co-exist with scripting, unlike rules which cannot.
